full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -4,589 +4,1000 @@
+
+        expect(content!.contentSet.length).to.eq(4);
+      });
+
+      it("uses `instanceFilter` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.InstanceFilter.Ruleset
+        // The specification returns content of all filtered `bis.SpatialViewDefinition` instances
+        // whose `Pitch` property value is higher or equal to 0.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["SpatialViewDefinition"] },
+              instanceFilter: "this.Pitch >= 0",
+            }],
           }],
-          propertyOverrides: [{
-            name: "*",
-            categoryId: "custom-category",
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that only `bis.SpatialViewDefinition` instances that have Pitch >= 0 are selected.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+
+        expect(content!.contentSet.length).to.eq(2);
+        const field = getFieldByLabel(content!.descriptor.fields, "Pitch");
+        content!.contentSet.forEach((record) => {
+          expect(record.values[field.name]).to.be.not.below(0);
+        });
+      });
+
+      it("uses `onlyIfNotHandled` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.OnlyIfNotHandled.Ruleset
+        // Specifications to return content for `bis.ViewDefinition` and `bis.PhysicalModel` respectively.
+        // The `bis.PhysicalModel` specification is lower priority and has `onlyIfNotHandled` attribute, which
+        // allows it to be overriden by higher priority specification.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["ViewDefinition"], arePolymorphic: true },
+            }, {
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["PhysicalModel"], arePolymorphic: true },
+              onlyIfNotHandled: true,
+            }],
           }],
-        }],
-      };
-      // __PUBLISH_EXTRACT_END__
-
-      // Ensure all `bis.GeometricElement3d` properties are in the custom category
-      const content = await Presentation.presentation.getContent({
-        imodel,
-        rulesetOrId: ruleset,
-        keys: new KeySet([{ className: "BisCore:Element", id: "0x61" }]),
-        descriptor: {},
+        };
```

Comment:
```suggestion
        // This ruleset defines two specifications that return content for `bis.ViewDefinition` and `bis.PhysicalModel`
        // instances respectively
        const ruleset: Ruleset = {
          id: "example",
          rules: [{
            ruleType: RuleTypes.Content,
            specifications: [
              {
                specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
                classes: { schemaName: "BisCore", classNames: ["ViewDefinition"], arePolymorphic: true },
              },
              // The following speficication is lower in priority and has `onlyIfNotHandled` attribute, which allows the
              // specification to be overriden by the specification above
              {
                specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
                classes: { schemaName: "BisCore", classNames: ["PhysicalModel"], arePolymorphic: true },
                onlyIfNotHandled: true,
              },
            ],
          }],
        };
```

Reply:
The suggestion to move some comments next to the specification is great. However:
- [ ] There's a typo: "The following **speficication** is lower".
- [ ] I suppose content request should stay?
- [ ] When reading "specification is lower in priority" I would raise a question "why?", because there's no priority attribute that would set it. Maybe we should add "due to definition order" or something like that?